frontend/app-development/features/processEditor/handlers/OnProcessTaskRemoveHandler.ts

```diff
@@ -1,26 +1,75 @@
 import type { Policy } from 'app-shared/types/Policy';
 import type { OnProcessTaskEvent } from '@altinn/process-editor/types/OnProcessTask';
 import { PaymentPolicyBuilder } from '../../../utils/policy';
+import {
+  getDataTypeIdFromBusinessObject,
+  getLayoutSetIdFromTaskId,
+} from '@altinn/process-editor/utils/hookUtils/hookUtils';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
 
 export class OnProcessTaskRemoveHandler {
   constructor(
     private readonly org: string,
     private readonly app: string,
     private readonly currentPolicy: Policy,
+    private readonly layoutSets: LayoutSets,
     private readonly mutateApplicationPolicy: (policy: Policy) => void,
+    private readonly deleteDataTypeFromAppMetadata: (data: { dataTypeId: string }) => void,
+    private readonly deleteLayoutSet: (data: { layoutSetIdToUpdate: string }) => void,
   ) {}
 
   /**
    * This handler is responsible to react on task remove event, to mutate files that are related to the task, but not a part of process-editor domain.
    * @param taskMetadata
    */
   public handleOnProcessTaskRemove(taskMetadata: OnProcessTaskEvent): void {
+    if (taskMetadata.taskType === 'data') {
+      this.handleDataTaskRemove(taskMetadata);
+    }
+
     if (taskMetadata.taskType === 'payment') {
       this.handlePaymentTaskRemove(taskMetadata);
     }
+
+    if (taskMetadata.taskType === 'signing') {
+      this.handleSigningTaskRemove(taskMetadata);
+    }
+  }
+
+  /**
+   * Deletes the layoutSet from the deleted data task
+   * @private
+   */
+  private handleDataTaskRemove(taskMetadata: OnProcessTaskEvent): void {
+    const layoutSetId = getLayoutSetIdFromTaskId(
+      taskMetadata.taskEvent.element.id,
+      this.layoutSets,
+    );
+
+    if (layoutSetId) {
+      this.deleteLayoutSet({
+        layoutSetIdToUpdate: layoutSetId,
+      });
+    }
   }
 
+  /**
+   * Deletes the dataType, layoutSet and policy for the deleted payment task
+   * @param taskMetadata
+   * @private
+   */
   private handlePaymentTaskRemove(taskMetadata: OnProcessTaskEvent): void {
+    // Delete dataType
```

Comment:
Is this comment needed? 😄 Same for line 72, 86, 105, and 115

Reply:
Since this method performs multiple tasks, I added comments to each "thing" to increase readability by creating a "visual section" and clarify what happens within the method.